test/e2e/e2eutil/common.go

```diff
@@ -198,6 +210,21 @@ func AssertUnstructuredExists(ctx context.Context, c client.Client, obj *unstruc
 	return resultObj
 }
 
+func AssertHasDeletionTimestamp(ctx context.Context, c client.Client, obj *unstructured.Unstructured) *unstructured.Unstructured {
+	ref := mutation.ResourceReferenceFromUnstructured(obj)
+	resultObj := ref.ToUnstructured()
+
+	err := c.Get(ctx, types.NamespacedName{
+		Namespace: obj.GetNamespace(),
+		Name:      obj.GetName(),
+	}, resultObj)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred(),
+		"expected GET not to error (%s): %s", ref, err)
+	gomega.Expect(resultObj.GetDeletionTimestamp()).NotTo(gomega.BeNil(),
```

Comment:
i dont know if deletion timestamp is always nil after parsing. there's an IsZero func I think. The k8s timestamp is a weird wrapper.

Reply:
done